### PR TITLE
mkplot with missing nuisances

### PR DESCRIPTION
**Pull request by amassiro**
Possibility to skip nuisances if missing.

Default is "FALSE", than nothing changes.

To trigger this option in mkPlot.py do

    --SkipMissingNuisance

@valsdav to test before merging, since you asked for this :)

The same for mkdatacards was already available

### CODE REVIEW

File: ShapeAnalysis/python/PlotFactory.py [hunk 1]
```diff
@@ -458,7 +458,17 @@ def makePlot(self, inputFile, outputDirPlots, variables, cuts, samples, plot, nu
                       else:
                         histoVar = fileIn.Get(shapeNameVar)
   
-                      nuisanceHistos[ivar][nuisanceName] = histoVar
+                      if histoVar != None :
+                        nuisanceHistos[ivar][nuisanceName] = histoVar
+                      elif not self._SkipMissingNuisances :
```
Comment (valsdav) [inline ShapeAnalysis/python/PlotFactory.py:463]: typo  SkipMissingNuisances --> skipMissingNuisances

File: ShapeAnalysis/python/PlotFactory.py [hunk 2]
```diff
@@ -458,7 +458,17 @@ def makePlot(self, inputFile, outputDirPlots, variables, cuts, samples, plot, nu
...
+                        print " This is bad, the nuisance ", nuisanceName, " is missing! You need to add it, maybe some jobs crashed?"
+                        nuisanceHistos[ivar][nuisanceName] = histoVar
+                      else :
+                        # if you had self._SkipMissingNuisances set to true, put the variation the same as the nominal
+                        histoVar = histo.Clone(shapeNameVar.replace('/', '__'))
+                        histoVar.Scale(variations[ivar])
```
Comment (valsdav) [inline ShapeAnalysis/python/PlotFactory.py:469]: variations is not defined here for shape uncertainties

File: ShapeAnalysis/python/PlotFactory.py (hunk 1 above)
Comment (amassiro) [reply ShapeAnalysis/python/PlotFactory.py:463]: ok

File: ShapeAnalysis/python/PlotFactory.py (hunk 2 above)
Comment (amassiro) [reply ShapeAnalysis/python/PlotFactory.py:469]: ok, no scaling is needed, thanks!

**Pull request merged**